packages/react-sdk/src/components/link.tsx

```diff
@@ -1,13 +1,51 @@
+import { Link as RemixLink, useLocation } from "@remix-run/react";
 import { forwardRef, type ElementRef, type ComponentProps } from "react";
 
+const isAbsoluteUrl = (href: string) => {
+  // http:, mailto:, tel:, etc.
+  return /^[a-z]+:/i.test(href);
+};
+
+// @todo: Ideally this code should live in designer app.
+// Sdk should not know how routing is implemented in designer.
+// We can move it there later when we have a way of customizing components per environment.
+const preserveMode = (href: string, currentSearch: string) => {
+  if (isAbsoluteUrl(href)) {
+    return href;
+  }
+
+  const mode = new URLSearchParams(currentSearch).get("mode");
+
+  if (mode !== "preview" && mode !== "edit") {
+    return href;
+  }
+
+  const [path, search] = href.split("?");
```

Comment:
```js
const url = new URL(href, fakeBaseIfNotAbsolute)
url.searchParams.set()
return isAbs ? url.href : url.path...
```

Reply:
Do you think it's bug prone to `href.split("?")`? Because otherwise current way seem simpler to me